karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/PipelineRoleDeployer.java

```diff
@@ -42,7 +42,7 @@ public Role desired(Karavan karavan, Context<Karavan> context) {
                 .withRules(
                         new PolicyRuleBuilder().withApiGroups("").withResources("secrets", "configmaps", "services", "persistentvolumes", "persistentvolumeclaims").withVerbs("*").build(),
                         new PolicyRuleBuilder().withApiGroups("networking.k8s.io").withResources("ingresses").withVerbs("*").build(),
-                        new PolicyRuleBuilder().withApiGroups("route.openshift.io").withResources( "routes").withVerbs("*").build(),
+                        new PolicyRuleBuilder().withApiGroups("route.openshift.io").withResources( "routes").withResources( "routes/custom-host").withVerbs("*").build(),
```

Comment:
What "routes/custom-host" mean?

Reply:
Probably it should be omitted, this is additional permission to set `host: def.abc.com` field.
However, to provide customization ability of the `service/route/ingress` objects, corresponding fields should be specified in `Karavan` CRD in hierarchical manner, not plain one.